src/activitypub/mod.rs

```diff
@@ -52,6 +91,43 @@ impl AsActivityPub for Account {
             "preferredUsername": self.username,
             "name": self.display_name.as_ref().map(String::as_str).unwrap_or(""),
             "summary": self.summary.as_ref().map(String::as_str).unwrap_or("<p></p>"),
-        }))
+        })))
+    }
+}
+
+impl AsActivityPub for Status {
+    fn as_activitypub(
+        &self,
+        conn: &db::Connection,
+    ) -> Result<ActivityStreams<serde_json::Value>, Error> {
+        Ok(ActivityStreams(json!({
+            "@context": ["https://www.w3.org/ns/activitystreams", {"sensitive": "as:sensitive"}],
+            "type": "Note",
+            "id": self.get_uri(conn)?,
+            "attributedTo": self.account(conn)?.get_uri(),
+
+            "content": self.text,
+            "summary": self.content_warning,
+            "sensitive": self.content_warning.is_some(),
+            "published": self.created_at.to_rfc3339(),
+        })))
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn identifies_ap_requests() {
```

Comment:
this covers just the guard function, right? is it possible to write DB tests yet?

Reply:
correct, yes
and i think it's possible. @yipdw and i discussed this once